### PR TITLE
Validation warnings

### LINKED ISSUE #327: Subrecipients: Don't block uploads when subrecipient details don't match
Let's aim for parity with CARES reporter:

- First upload that specifies a Subrecipient is captured
- Duplicate Subrecipients in subsequent reports are ignored
- (Optional) Report simplified "warnings" during the upload process

### PR DESCRIPTION
**Pull request by igor47**
fixes #327 

we display mismatch errors as warnings in the UI. we still treat missing fields as fatal errors.

### CODE REVIEW

File: src/server/lib/validation-error.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 class ValidationError extends Error {
-  constructor (message, { severity = 1, tab = null, row = null, col = null } = {}) {
+  constructor (message, { severity = 'err', tab = null, row = null, col = null } = {}) {
```
Comment (ajhyndman) [inline src/server/lib/validation-error.js:3]: Nit: I like using descriptive strings over numbers, but I also have a pet peeve for unnecessary abbreviations.  Could we just call it `'error'`?

File: src/server/services/validate-upload.js [hunk 2]
```diff
@@ -31,9 +31,10 @@ async function validateAgencyId ({ upload, records, trns }) {
     )
   }
 
-  // set agency id on the upload, for disambiguation
+  // always set agency id if possible; we omit passing the transaction in this
+  // case, so that the agency id gets set even if the upload fails to validate
   if (matchingAgency.id !== upload.agency_id) {
-    await setAgencyId(upload.id, matchingAgency.id, trns)
+    await setAgencyId(upload.id, matchingAgency.id)
```
Comment (ajhyndman) [inline src/server/services/validate-upload.js:37]: Just double-checking.  If the validation errors immediately above are triggered this should not run, correct?

File: src/server/lib/validation-error.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 
 class ValidationError extends Error {
```
Comment (ajhyndman) [inline src/server/lib/validation-error.js:2]:
```suggestion
/**
 * severity: "err" | "warn"
 */
class ValidationError extends Error {
```

Could you add a short comment or something expressing which severities are expected?

**Review by ajhyndman — APPROVED** [review]
A couple minor requests, but looks good!

File: src/server/lib/validation-error.js (hunk 1 above)
Comment (igor47) [reply src/server/lib/validation-error.js:3]: i'm mirroring what i did in https://github.com/usdigitalresponse/arpa-reporter/blob/3a3851f2f31b5580f1b86771424d6ecd111c61c6/src/components/AlertBox.vue#L20

File: src/server/services/validate-upload.js (hunk 2 above)
Comment (igor47) [reply src/server/services/validate-upload.js:37]: yes, those immediately `return` with the error. we only get here if there was no error.

**Pull request merged**